Lib/test/test_float.py

```diff
@@ -700,12 +700,20 @@ def test_format(self):
         # % formatting
         self.assertEqual(format(-1.0, '%'), '-100.000000%')
 
+        # hexadecimal format
+        self.assertEqual(format(float.fromhex('0x0.0030p+0'), 'x'), '0x1.8p-11')
+        self.assertEqual(format(float.fromhex('0x0.0040p+0'), 'x'), '0x1p-10')
```

Comment:
I suggest adding `num = float.fromhex('0x0.0040p+0')` and then use *num*, rather than copy/paste it on each line.